### DISCUSSION TITLE
[Question]any try on fastapi as backend service

### DISCUSSION THREAD

**Discussion by wanghaisheng** [flet-dev/flet#1446] [Q&A]
### Question

fastapi is popular and i was wondering if there is any advice or existing work I can learn from

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by gbrandt** [comment]
This does not seem like a flet issue at all

**Reply by wanghaisheng** [reply]
you can say that.if I want to user flet as main framework to develop multi platform app, I think many have the same issue,how to manage user for example, so it came to this question

**Comment by bfrench4255** [comment]
As mentioned earlier, its not a Flet issue, but you can use fastapi within Flet to retrieve data for display within the application. You will have a function within a Flet page that calls the api using Python requests library against a fastapi endpoint. You would then use the data (JSON for example) returned from the endpoint to display within Flet i.e., a DataTable for example.

**Reply by wanghaisheng** [reply]
this is very confused.
how could this be?
does  not desktop app  need database any  chance?
if it is true, would there be any doc about how to connect database or mature and popular backend service framework?

**Comment by DmitryYalchik** [comment]
> does not desktop app need database any chance? if it is true, would there be any doc about how to connect database or mature and popular backend service framework?

Almost all APIs use databases. For example: u can use SQLite3. There are a lot of videos on youtube about this. Just try to search it before opening issue 😃

**Reply by wanghaisheng** [reply]
I  did not see any flet doc around integrations with database

**Reply by FeodorFitsner** [reply]
Flet program is just a Python program - you can use any library for working with database, such as SQLAlchemy.

**Reply by wanghaisheng** [reply]
I definitely know that, all i want is a clear documentation about this topic  unless you guys think this is totally  useless

**Reply by DmitryYalchik** [reply]
Programming is like constructor 🏗️
Just find some video-/text tutorial about how include database in ur python project and follow steps in ur flet project.

U can do new tutorial about this. It's can to help u to learn new tech better 😃
